ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/BesuPrivateNonceProvider.java

```diff
@@ -12,54 +12,42 @@
  */
 package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
 
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequestId;
 import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
 
-import java.io.IOException;
 import java.math.BigInteger;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.web3j.protocol.besu.Besu;
-import org.web3j.protocol.core.Request;
-import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
 import org.web3j.utils.Base64String;
 
 public class BesuPrivateNonceProvider implements NonceProvider {
 
-  private static final Logger LOG = LogManager.getLogger();
-
-  private final Besu besu;
+  protected static int lastId = 0;
```

Comment:
Thinking there that the lastId should be in central place so that ids don't overlap too much. Perhaps move the incrementing and setting of id into the VertxNonceRequestTransmitter

Reply:
Actually - that'd work quite nicely...